### PR TITLE
VMware: vmware_portgroup teaming policy support

### LINKED ISSUE #36773: unable to specify NIC teaming type while creating port group on vswitch on ESXI host
##### ISSUE TYPE

 - Bug Report
 

##### COMPONENT NAME

```
vmware_portgroup
```
##### ANSIBLE VERSION

```
2.5.0a1
```

##### SUMMARY

```
  while creating portgroup on vswitch i want specify NIC teaming policy but current module implementation doesnt have that facility, same facility is present for vmware_dvs_portgroup module
```
##### STEPS TO REPRODUCE

```yaml

```

##### ACTUAL RESULTS

```

```

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
This fix adds support for specifying teaming policy in
vmware_portgroup module.

Fixes: #36773

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
lib/ansible/module_utils/vmware.py
lib/ansible/modules/cloud/vmware/vmware_portgroup.py

##### ANSIBLE VERSION

```
2.6devel
```

### CODE REVIEW

**Comment by Akasurde** [comment]
@pdellaert @tchernomax Could you please review this ?

File: lib/ansible/module_utils/vmware.py [hunk 1]
```diff
@@ -1140,10 +1142,14 @@ def get_all_host_objs(self, cluster_name=None, esxi_host_name=None):
...
             elif esxi_host_name:
-                esxi_host_obj = self.find_hostsystem_by_name(host_name=esxi_host_name)
-                if esxi_host_obj:
-                    host_obj_list = [esxi_host_obj]
-                else:
-                    self.module.fail_json(changed=False, msg="ESXi '%s' not found" % esxi_host_name)
+                if not isinstance(esxi_host_name, list):
+                    esxi_host_name = [esxi_host_name]
+
+                for host in esxi_host_name:
```
Comment (pdellaert) [inline lib/ansible/module_utils/vmware.py]: Isn't this code going to be executed even if `esxi_host_name` is not a list? Causing an exception or unexpected behaviour?

File: lib/ansible/modules/cloud/vmware/vmware_portgroup.py [hunk 2]
```diff
@@ -285,36 +319,33 @@ def create_host_port_group(self, host_system, portgroup_name, vlan_id, vswitch_n
...
-                except vim.fault.HostConfigFault as host_config_fault:
-                    self.module.fail_json(msg="Failed to update Portgroup due to host"
-                                              " system configuration failure : %s" % to_native(host_config_fault.msg))
-                except vmodl.fault.InvalidArgument as invalid_argument:
-                    self.module.fail_json(msg="Failed to update Portgroup as VLAN id was not"
-                                              " correct as per specifications: %s" % to_native(invalid_argument.msg))
-                except Exception as generic_exception:
-                    self.module.fail_json(msg="Failed to update Portgroup due to generic"
-                                              " exception : %s" % to_native(generic_exception))
+            try:
```
Comment (pdellaert) [inline lib/ansible/modules/cloud/vmware/vmware_portgroup.py]: This means you are calling the `UpdatePortGroup` every time you run the module? Even if the state has not changed?

Comment (Akasurde) [reply lib/ansible/modules/cloud/vmware/vmware_portgroup.py]: Yes. I have seen weird behaviour while calling `UpdatePortGroup` API like [this](https://github.com/ansible/ansible/issues/36773#issuecomment-369659095) So I am calling it every time. Any suggestions ?

File: lib/ansible/module_utils/vmware.py (hunk 1 above)
Comment (Akasurde) [reply lib/ansible/module_utils/vmware.py]: I didn't follow you here.

Comment (pdellaert) [reply lib/ansible/module_utils/vmware.py]: Nevermind, i missed the fact that you force `esxi_host_name` to be a list, sorry

File: lib/ansible/modules/cloud/vmware/vmware_portgroup.py (hunk 2 above)
Comment (pdellaert) [reply lib/ansible/modules/cloud/vmware/vmware_portgroup.py]: not sure about bug, but i just wonder why you would call the `UpdatePortGroup` if `self.changed == False` (now, you even call it if nothing changed). You are updating something that doesn't need an update?

Comment (Akasurde) [reply lib/ansible/modules/cloud/vmware/vmware_portgroup.py]: I tried on my setup, not calling update API if nothing is changed. Tweaked code a bit. Thanks for suggestion.

**Review by pdellaert — APPROVED** [review]
LGTM

**Comment by Akasurde** [comment]
rebuild_merge

**Comment by Akasurde** [comment]
@pdellaert Thanks for reviews.

**Pull request merged**